packages/ckeditor5-dev-license-checker/src/validate-license-files.ts

```diff
@@ -0,0 +1,329 @@
+ * @param options.isPublic Whether license should use disclaimer meant for open source repositories.
+ * @param options.rootDir Base directory.
+ * @param options.projectName Project name referred to in the licenses.
+ * @param options.mainPackageName Designated package that contains collected licenses from all other packages.
+ * @param options.copyrightOverrides Map of of copyright that can both add new ones, as well as override existing ones.
+ *
+ * @returns Exit code of the script that indicates whether it passed or errored.
+ */
+export async function validateLicenseFiles( {
+	fix = false,
+	processRoot = false,
+	processPackages = false,
+	isPublic = false,
+	rootDir,
+	projectName,
+	mainPackageName,
+	copyrightOverrides = []
+}: {
+	fix?: boolean;
+	processRoot?: boolean;
+	processPackages?: boolean;
+	isPublic?: boolean;
+	rootDir: string;
+	projectName: string;
+	mainPackageName?: string;
+	copyrightOverrides?: Array<CopyrightOverride>;
+} ): Promise<number> {
+	const packagePaths: Array<string> = [];
+
+	if ( processRoot ) {
+		packagePaths.push( rootDir );
+	}
+
+	if ( processPackages ) {
+		packagePaths.push( ...await fromAsync( glob( upath.join( rootDir, 'packages', '*' ) ) ) );
+	}
+
+	if ( !processRoot && !processPackages ) {
+		console.error( 'You have to set at least one of: `processRoot` or `processPackages`.' );
+
+		return 1;
+	}
+
+	// Collect versioning and licensing data of all dependencies.
+	const dependencyMap = await Promise.all( packagePaths.map( async packagePath => {
+		const pkgJsonPath = upath.join( packagePath, 'package.json' );
+		const pkgJsonContent = JSON.parse( await readFile( pkgJsonPath, 'utf-8' ) );
+		const dependencyNames = Object.keys( pkgJsonContent.dependencies || {} )
+			.filter( dependency => !dependency.match( /(ckeditor)|(cksource)/i ) );
+
+		const packageName = pkgJsonContent.name;
+
+		const dependencyMapItem: DependencyMapItem = {
+			packageName,
+			packagePath,
+			dependencies: []
+		};
+
+		const copyrightOverridesPackage = copyrightOverrides
+			.find( ( { packageName: overridePackageName } ) => packageName === overridePackageName );
+
+		if ( copyrightOverridesPackage ) {
+			dependencyMapItem.dependencies.push( ...copyrightOverridesPackage.dependencies );
+		}
+
+		for ( const dependencyName of dependencyNames ) {
+			// If override already exists, skip parsing the dependency.
+			if ( dependencyMapItem.dependencies.some( ( { name } ) => name === dependencyName ) ) {
+				continue;
+			}
+
+			const dependencyPath = upath.join( packagePath, 'node_modules', dependencyName );
+			const dependencyPkgJsonPath = upath.join( dependencyPath, 'package.json' );
+			const dependencyPkgJsonContent = JSON.parse( await readFile( dependencyPkgJsonPath, 'utf-8' ) );
+
+			dependencyMapItem.dependencies.push( {
+				name: dependencyName,
+				license: dependencyPkgJsonContent.license,
+				copyright: await getCopyright( dependencyPath )
```

Comment:
Just a thought, but how much overriding in the `validate-license.mjs` script would we need if we decided to just read the `author` field from `package.json` rather than parse the `LICENSE.md` file?

I know that the output files would be slightly different, but maybe not _worse_.